include/swift/SILOptimizer/Utils/Generics.h

```diff
@@ -52,20 +52,78 @@ class ReabstractionInfo {
   /// The first NumResults bits in Conversions refer to indirect out-parameters.
   unsigned NumResults;
 
-  /// The function type after applying the substitutions of the original
-  /// apply site.
+  /// The function type after applying the substitutions used to call the
+  /// specialized function.
   CanSILFunctionType SubstitutedType;
 
   /// The function type after applying the re-abstractions on the
   /// SubstitutedType.
   CanSILFunctionType SpecializedType;
 
+  /// The generic environment to be used by the specialization.
+  GenericEnvironment *SpecializedGenericEnv;
```

Comment:
I see. I'll try to move SIL generation changes into a subsequent patch. Eventually, I'll have to disable the EagerSpecializer temporarily, because it needs to use the new @_specialize attribute for the codegen, but won't be able to do it without Generics.cpp changes.